### DISCUSSION TITLE
eager load the lucid relation with the custom field

### DISCUSSION THREAD

**Discussion by farshadfahimi** [adonisjs/core#1746] [Help]
I have a simple class like below and have a relation for fetch the `prices` of products but it return the null for me in eager loading otherwise when I'm using `fetch` for get the relation data it return data for me.

```javascript
class Product {
  public function price() {
     return this.hasOne('App/Models/Prices', 'id', 'test_prices.products_id')
  }
}
```

while I'm trying to defined the problem. I found when in the relation I remove `test_prices` it work fine in the eager loading.
but when add the table name before it it's not work.

How I should resolve this?

**Comment by JaconsMorais** [comment]
Hi, you need to use database.raw when defining table.field, try this:


```
const database = use('Database')

class Product {
  public function price() {
     return this.hasOne('App/Models/Prices', 'id', database.raw('test_prices.products_id'))
  }
}
```
